receiver/couchbasereceiver/factory.go

```diff
@@ -0,0 +1,59 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package couchbasereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchbasereceiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+)
+
+const typeStr = "couchbase"
+
+// NewFactory creates a new receiver factory
+func NewFactory() component.ReceiverFactory {
+	return receiverhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiverhelper.WithMetrics(createMetricsReceiver))
+}
+
+func createDefaultConfig() config.Receiver {
+	return &Config{
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			ReceiverSettings:   config.NewReceiverSettings(config.NewComponentID(typeStr)),
+			CollectionInterval: 10 * time.Second,
+		},
+		HTTPClientSettings: confighttp.HTTPClientSettings{
+			TLSSetting: configtls.TLSClientSetting{
+				Insecure:           false,
+				InsecureSkipVerify: true,
```

Comment:
I'm not sure that these are appropriate default settings.  Having a configuration that is insecure by default seems like a recipe for a CVE.

Reply:
If the user specifies a non-TLS `http://` endpoint, that's on them.  If they specify to use TLS with an `https://` endpoint, we should not make that configuration insecure by default.  The user can further choose to disable certificate validation, but that's on them.

Reply:
That makes sense. I think because the default endpoint has `http://` we should use the zero value `TLSClientSetting` which would have `Insecure` and `InsecureSkipVerify` as false. 

Does that seem like a fair approach or is there more to this that I'm missing?

Reply:
That sounds good to me.

Reply:
@Aneurysm9 Do you mind giving this PR another look now that I've updated the TLS config?

Reply:
Oh sorry I just saw later in the comment stake you dismissed your concerns. Thanks for your time!